src/coreclr/tests/src/Loader/binding/tracing/BinderTracingTest.DefaultProbing.cs

```diff
@@ -0,0 +1,214 @@
+                AssemblyLoadContext = DefaultALC,
+                RequestingAssembly = Assembly.GetExecutingAssembly().GetName(),
+                RequestingAssemblyLoadContext = DefaultALC,
+                Success = false,
+                Cached = false,
+                ProbedPaths = new List<ProbedPath>()
+                {
+                    new ProbedPath()
+                    {
+                        FilePath = Helpers.GetProbingFilePath(ProbedPath.PathSource.AppNativeImagePaths, assemblyName, isExe: false),
+                        Source = ProbedPath.PathSource.AppNativeImagePaths,
+                        Result = COR_E_FILENOTFOUND
+                    },
+                    new ProbedPath()
+                    {
+                        FilePath = Helpers.GetProbingFilePath(ProbedPath.PathSource.AppNativeImagePaths, assemblyName, isExe: true),
+                        Source = ProbedPath.PathSource.AppNativeImagePaths,
+                        Result = COR_E_FILENOTFOUND
+                    },
+                    new ProbedPath()
+                    {
+                        FilePath = Helpers.GetProbingFilePath(ProbedPath.PathSource.AppPaths, assemblyName, isExe: false),
+                        Source = ProbedPath.PathSource.AppPaths,
+                        Result = COR_E_FILENOTFOUND
+                    },
+                    new ProbedPath()
+                    {
+                        FilePath = Helpers.GetProbingFilePath(ProbedPath.PathSource.AppPaths, assemblyName, isExe: true),
+                        Source = ProbedPath.PathSource.AppPaths,
+                        Result = COR_E_FILENOTFOUND
+                    }
+                }
+            };
+        }
+
+        [BinderTest(isolate: true)]
+        public static BindOperation SatelliteAssembly_AppPath()
+        {
+            AssemblyName assemblyName = new AssemblyName($"{DependentAssemblyName}.resources");
+            assemblyName.CultureInfo = SatelliteCulture;
+            Assembly asm = AssemblyLoadContext.Default.LoadFromAssemblyName(assemblyName);
+
+            return new BindOperation()
+            {
+                AssemblyName = assemblyName,
+                AssemblyLoadContext = DefaultALC,
+                Success = true,
+                ResultAssemblyName = asm.GetName(),
+                ResultAssemblyPath = asm.Location,
+                Cached = false,
+                ProbedPaths = new List<ProbedPath>()
+                {
+                    new ProbedPath()
+                    {
+                        FilePath = Helpers.GetProbingFilePath(ProbedPath.PathSource.AppPaths, assemblyName.Name, SatelliteCulture.Name),
+                        Source = ProbedPath.PathSource.AppPaths,
+                        Result = S_OK
+                    }
+                }
+            };
+        }
+
+        [BinderTest]
+        public static BindOperation SatelliteAssembly_CultureSubdirectory()
```

Comment:
While you're at it - can you add a test to validate that on Linux we perform case sensitive probe and then lower cased probe?